indigo_api/management/commands/migrate_taxonomies.py

```diff
@@ -0,0 +1,24 @@
+from django.core.management import BaseCommand
+from indigo_api.models import TaxonomyTopic, TaxonomyVocabulary, Work
+import logging
+
+log = logging.getLogger(__name__)
+
+class Command(BaseCommand):
+    def handle(self, *args, **kwargs):
+        for t in TaxonomyVocabulary.objects.all().iterator():
+            root = TaxonomyTopic.add_root(name=t.title)
+            log.info(f"Created root node {root}")
+            for vocabulary in t.topics.all():
+                if vocabulary.level_1:
+                    log.info(f"Creating child node {vocabulary.level_1}")
+                    child_1 = TaxonomyTopic.add_child(root, name=vocabulary.level_1)
+                    for work in vocabulary.works.all():
```

Comment:
I think you're over-thinking this. If `vocabulary` has a `level_2`, then the works are tagged at the second level. Otherwise, they're tagged at the first level.

See here how `level_1` is duplicated when there is a `level_2`:

![image](https://github.com/laws-africa/indigo/assets/4178542/6ec37c10-b85b-4211-b499-e21aaff97d9c)

So you only need to apply EITHER `level_2` (if it's not blank) OR fall back to `level_1`. Not both.

Reply:
Updated :+1: